### PR TITLE
will/slider

### PR DESCRIPTION
**Pull request by WillHHH**

### CODE REVIEW

File: src/components/Slider/index.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import classnames from '../../utils/classNames';
+import { IProps } from './types';
+
+import './index.scss';
+
+const baseClass = 'vant-slider';
+
+const Slider = ({
+  range = { min: 0, max: 100 },
+  size = { width: 400, height: 5 },
```
Comment (mxdi9i7) [inline src/components/Slider/index.tsx:12]: the value here should have unit, like `400px`

File: src/components/Slider/index.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  sliderSize = { width: 20, height: 20 },
```
Comment (mxdi9i7) [inline src/components/Slider/index.tsx:13]: same as above

File: src/components/Slider/index.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+          dom3.style.width = `${
+            e.pageX - dom1.offsetLeft - sliderOffset + HIDE_ROUND
+          }px`;
+          setValue(getValue(e, dom1));
+        }
+      }
+    });
+  };
+
+  if (disabled) {
```
Comment (mxdi9i7) [inline src/components/Slider/index.tsx:123]: should just assign the disabled class to the parent, and use css to apply to children.

File: src/components/Slider/types.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+export interface IProps {
+  disabled?: boolean;
+  hasValue?: boolean;
+  activeColor?: string;
+  inactiveColor?: string;
+  size?: { width: number; height: number };
```
Comment (mxdi9i7) [inline src/components/Slider/types.ts:6]: size should be string, with unit

File: src/components/Slider/types.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  sliderStyle?: {
+    color: string;
+    fontSize: string;
+    backgroundColor: string;
+    borderRadius: string;
+    borderColor: string;
+  };
+  range?: { min: number; max: number };
+  id?: string;
+  value: number | undefined;
```
Comment (mxdi9i7) [inline src/components/Slider/types.ts:17]: how can it be undefined?

File: src/components/Slider/index.tsx (hunk 1 above)
Comment (WillHHH) [reply src/components/Slider/index.tsx:12]: fixed

File: src/components/Slider/index.tsx (hunk 2 above)
Comment (WillHHH) [reply src/components/Slider/index.tsx:13]: fixed

File: src/components/Slider/types.ts (hunk 4 above)
Comment (WillHHH) [reply src/components/Slider/types.ts:6]: fixed

File: src/components/Slider/types.ts (hunk 5 above)
Comment (WillHHH) [reply src/components/Slider/types.ts:17]: fixed

File: src/components/Slider/index.tsx (hunk 3 above)
Comment (WillHHH) [reply src/components/Slider/index.tsx:123]: fixed

**Pull request merged**